src/vm_pages.rs

```diff
@@ -151,31 +158,201 @@ impl TlbTracker {
     }
 }
 
+/// Types of regions in a VM's guest physical address space.
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum VmRegionType {
+    /// Regions of private memory that have been fully populated during VM creation.
+    PreMapped,
+    /// Placeholder region type to reserve a range of guest physical address space while the region
+    /// is mapped.
+    Reserved,
+}
+
+/// A contiguous region of guest physical address space.
+struct VmRegion {
+    start: GuestPageAddr,
+    end: GuestPageAddr,
+    region_type: VmRegionType,
+}
+
+/// Represents a reserved region in `VmRegionList` that has yet to be committed. Removes the reserved
+/// region if dropped without being passed to `VmRegionList::commit()`.
```

Comment:
I would find it useful to have an example or basic flow overview of how/why this is used with the `prepare` and `commit` functions below.

Reply:
I have a very similar structure in my upcoming patch for page sharing....we can look into combining them in the future.

Reply:
Expanded the comment above `VmRegionList` with more details on why along with a simple example.